subsys/bluetooth/mesh/priv_beacon_srv.c

```diff
@@ -206,7 +207,13 @@ static int priv_beacon_srv_init(const struct bt_mesh_model *mod)
 	priv_beacon_srv = mod;
 	mod->keys[0] = BT_MESH_KEY_DEV_LOCAL;
 
-	bt_mesh_model_extend(mod, config_srv);
+	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
```

Comment:
Please revert this change. This is not necessary. Sometime ago we had discussion about this and the decision was to move this dependency from source code to KConfig file. 
Please take a look: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/Kconfig#L1266-L1270